packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js

```diff
@@ -0,0 +1,226 @@
+  _MEETING_LINK_,
+  _PEOPLE_,
+  _ROOM_,
+  DIALER_REGEX,
+  WEBEX_DOT_COM,
+  CONVERSATION_SERVICE,
+  JOIN,
+  MEET,
+  MEET_M,
+  HTTPS_PROTOCOL,
+  UUID_REG
+} from '../constants';
+import ParameterError from '../common/errors/parameter';
+import LoggerProxy from '../common/logs/logger-proxy';
+
+const MeetingInfoUtil = {};
+
+MeetingInfoUtil.getParsedUrl = (link) => {
+  try {
+    let parsedUrl = url.parse(link);
+
+    if (!parsedUrl) {
+      return false;
+    }
+    // hack for links such as <company>.webex.com/meet/<user> without a protocol
+    if (!parsedUrl.protocol) {
+      parsedUrl = url.parse(`${HTTPS_PROTOCOL}${link}`);
+    }
+
+    return parsedUrl;
+  }
+  catch (error) {
+    LoggerProxy.logger.warn(`Meeting-info:util#getParsedUrl --> unable to parse the URL, error: ${error}`);
+
+    return null;
+  }
+};
+
+/**
+ * Helper function to check if a string matches a known meeting link pattern
+ * @param {String} value  string to parse and see if it matches a meeting link
+ * @returns {Boolean}
+ */
+MeetingInfoUtil.isMeetingLink = (value) => {
+  const parsedUrl = MeetingInfoUtil.getParsedUrl(value);
+  const hostNameBool = parsedUrl.hostname && parsedUrl.hostname.includes(WEBEX_DOT_COM);
+  const pathNameBool = parsedUrl.pathname && (parsedUrl.pathname.includes(`/${MEET}`) || parsedUrl.pathname.includes(`/${MEET_M}`) || parsedUrl.pathname.includes(`/${JOIN}`));
+
+  return hostNameBool && pathNameBool;
+};
+
+MeetingInfoUtil.isConversationUrl = (value, webex) => {
+  const clusterId = webex.internal.services.getClusterId(value);
+
+  if (clusterId) {
+    return clusterId.endsWith(CONVERSATION_SERVICE);
+  }
+
+  return false;
+};
+
+
+MeetingInfoUtil.isSipUri = (sipString) => {
+  // TODO: lets remove regex from this equation and user URI matchers and such
+  // have not found a great sip uri parser library as of now
+  const sipUri = DIALER_REGEX.SIP_ADDRESS.exec(sipString);
+
+  return sipUri;
+};
+
+MeetingInfoUtil.isPhoneNumber = (phoneNumber) => {
+  const isValidNumber = DIALER_REGEX.PHONE_NUMBER.test(phoneNumber);
+
+  return isValidNumber;
+};
+
+MeetingInfoUtil.getHydraId = (destination) => {
+  const {type, id, cluster} = deconstructHydraId(destination);
+
+  if (id && UUID_REG.test(id)) {
+    if (type === _ROOM_) {
+      return {room: true, destination: id, cluster};
+    }
+    if (type === _PEOPLE_) {
+      return {people: true, destination: id, cluster};
+    }
+
+    return {};
+  }
+
+  return {};
+};
+
+MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.people.get(destination).then((res) => {
+  if (res.emails && res.emails.length) {
+    return res.emails[0];
+  }
+  throw new ParameterError('Hydra Id Lookup was an invalid hydra person id.');
+}).catch((err) => {
+  LoggerProxy.logger.error(`Meeting-info:util#MeetingInfoUtil.getSipUriFromHydraPersonId --> getSipUriFromHydraPersonId ${err} `);
+  throw err;
+});
+
+
+MeetingInfoUtil.getDestinationType = async (from) => {
```

Comment:
this function never returns a type of _PERSONAL_ROOM_ or _MEETING_ID_ or _LOCUS_ID_ and these types are handled in getRequestBody() - is this mismatch on purpose?

Reply:
Its a consequence of bad design .

Currently when we started the sdk we never asked the user to add the type with the destination . So we needed that switch statement 

If a user passed the type it will not go inside getDestinationType and the destination and type will be directly passed inside

Reply:
Need JSDoc comments for each method.